bin/python/ripple/util/Sign.py

```diff
@@ -0,0 +1,139 @@
+#!/usr/bin/env python
+
+from __future__ import print_function
+
+import base64, os, random, sys
+import ed25519
+import ecdsa
+from ripple.util import Base58
+from ripple.ledger import SField
+
+ED_BYTE = chr(0xed)
+
+USAGE = """\
+Usage:
+    create
+        Create a new master public/secret key pair.
+
+    check <key>
+        Check an existing key for validity.
+
+    sign <sequence> <validator-public> <master-secret>
+        Create a new signed manifest with the given sequence
+        number, validator public key, and master secret key.
+    """
+
+def to_bytes_bigend(n, length):
+    h = '%x' % n
+    return ('0' * (len(h) % 2) + h).zfill(length * 2).decode('hex')
+
+def prepend_length_byte(b):
+    assert len(b) < 192, 'Too long'
+    return chr(len(b)) + b
+
+#-----------------------------------------------------------
+
+def make_seed(urandom=os.urandom):
+    # This is not used.
+    return urandom(16)
+
+def make_ed25519_keypair(urandom=os.urandom):
+    private_key = urandom(32)
+    return private_key, ed25519.publickey(private_key)
+
+def make_ecdsa_keypair():
+    # This is not used.
+    private_key = ecdsa.SigningKey.generate(curve=ecdsa.SECP256k1)
+    # Can't be unit tested easily - need a mock for ecdsa.
+    vk = private_key.get_verifying_key()
+    sig = private_key.sign('message')
+    assert vk.verify(sig, 'message')
+    return private_key, vk
+
+def make_manifest(pk, vpk, seq):
+    return ''.join([
+        SField.sfSequence,
+        to_bytes_bigend(seq, 4),
+        SField.sfPublicKey,
+        prepend_length_byte(pk),
+        SField.sfSigningPubKey,
+        prepend_length_byte(vpk)])
+
+def sign_manifest(manifest, private_key, pk):
+    sig = ed25519.signature('MAN\0' + manifest, private_key, pk)
+    return manifest + SField.sfSignature + prepend_length_byte(sig)
+
+def wrap(s, cols=60):
+    if s:
+        size = max((len(s) + cols - 1) / cols, 1)
+        w = len(s) / size
+        s = '\n'.join(s[i:i + w] for i in range(0, len(s), w))
+    return s
+
+def create_ed_keys(urandom=os.urandom):
+    private_key, pk = make_ed25519_keypair(urandom)
+    apk = ED_BYTE + pk
+    pkh = Base58.encode_version(Base58.VER_NODE_PUBLIC, apk)
+    skh = Base58.encode_version(Base58.VER_NODE_PRIVATE, private_key)
+    v0, apk0 = Base58.decode_version(pkh)
+    assert v0 == Base58.VER_NODE_PUBLIC
+    assert apk0 == apk
+    return pkh, skh
+
+def create():
+    pkh, skh = create_ed_keys()
+    print('[validators]', pkh, '', '[master_secret]', skh, sep='\n')
+
+def check(s):
+    ver, b = Base58.decode_version(s)
+    print('version        = ' + Base58.version_name(ver))
+    print('decoded length = ' + str(len(b)))
+    h = Base58.encode_version(ver, b)
+    assert h == s
+
+def check_validator_public(v, avpk):
```

Comment:
@sublimator Yes. This tool is a temporary measure that just lets you sign manifests and generate keys. Eventually it will become part of the functionality in rsign.js (and include threshold signature features).